core/src/test/scala/akka/persistence/jdbc/DeleteTest.scala

```diff
@@ -0,0 +1,124 @@
+      |    snapshot-store {
+      |      plugin = "jdbc-snapshot-store"
+      |      // Enable the line below to automatically start the snapshot-store when the actorsystem is started
+      |      // auto-start-snapshot-stores = ["jdbc-snapshot-store"]
+      |    }
+      |  }
+      |}
+      |
+      |jdbc-journal {
+      |  slick = ${slick}
+      |}
+      |
+      |# the akka-persistence-snapshot-store in use
+      |jdbc-snapshot-store {
+      |  slick = ${slick}
+      |}
+      |
+      |# the akka-persistence-query provider in use
+      |jdbc-read-journal {
+      |  slick = ${slick}
+      |}
+      |
+      |slick {
+      |  profile = "slick.jdbc.H2Profile$"
+      |  db {
+      |    url = "jdbc:h2:mem:test-database;DATABASE_TO_UPPER=false;"
+      |    user = "root"
+      |    password = "root"
+      |    driver = "org.h2.Driver"
+      |    numThreads = 5
+      |    maxConnections = 5
+      |    minConnections = 1
+      |  }
+      |}
+      |""".stripMargin
+
+
+  case class Persist(value: String)
+  case class Delete(toSeq: Long)
+  case class Event(value: String)
+  case object AckPersist
+  case class AckDelete(toSeq: Long)
+
+  class PersistingActor extends PersistentActor {
+    private var deleteSender: ActorRef = _
+
+    override def persistenceId: String = "testId"
+
+    override def receiveRecover: Receive = {
+      case _ =>
+    }
+
+    override def receiveCommand: Receive = {
+      case Persist(value) =>
+        persist(Event(value)) { _ =>
+          sender() ! AckPersist
+        }
+
+      case Delete(toSeq) =>
+        deleteMessages(toSeq)
+        deleteSender = sender()
+
+      case m: DeleteMessagesSuccess =>
+        deleteSender ! AckDelete(m.toSequenceNr)
+    }
+  }
+}
+
+class DeleteTest private(_system: ActorSystem) extends TestKit(_system) with AnyFlatSpecLike with Matchers {
+  import DeleteTest._
+
+  def this() = this(
+    ActorSystem("h2test",
+      ConfigFactory.systemProperties()
+          .withFallback(ConfigFactory.parseString(DeleteTest.config))
+          .withFallback(ConfigFactory.load())
+          .resolve())
+  )
+
+  it should "delete events correctly" in {
+    SchemaUtils.createIfNotExists(system)
+
+    val probe = TestProbe()
+    val persister = system.actorOf(Props(new PersistingActor()))
+
+    probe.send(persister, Persist("1"))
+    probe.expectMsg(AckPersist)
+    probe.send(persister, Persist("2"))
+    probe.expectMsg(AckPersist)
+    probe.send(persister, Persist("3"))
+    probe.expectMsg(AckPersist)
+
+    probe.send(persister, Delete(2))
+    probe.expectMsg(AckDelete(2)) // should delete
+
+    val eventProbe = TestProbe()
+    PersistenceQuery(system).readJournalFor[JdbcReadJournal]("jdbc-read-journal").eventsByPersistenceId("testId", 0, Long.MaxValue).runForeach(e => eventProbe.ref ! e)
+    val event = eventProbe.expectMsgType[EventEnvelope]
+    event.sequenceNr shouldBe 3 // succeeds with Cassandra or LevelDB, with H2 event seqNr 2 is returned
```

Comment:
Excellent finding. Can you add the fix to this PR?

Reply:
The "fix" I found was just me poking around. 
I didn't go deep into the code of this project and honestly I have no idea if that is the correct way to fix it ;)

Reply:
I looked closer at this. There is already a filter in the `baseQuery`
https://github.com/akka/akka-persistence-jdbc/blob/068f0f0cc10db8fb38c4295bfa1397fd8e3ab87f/core/src/main/scala/akka/persistence/jdbc/query/dao/ReadJournalQueries.scala#L28-L29

Note the check of the config property for including deleted or not. By default it is using soft deletes, which was a surprise to me. I see that logicalDeletion has been deprecated but it is still true by default. Created issue https://github.com/akka/akka-persistence-jdbc/issues/557

Can you test again but set that config to false.

The purpose of keeping the last row is to know the latest sequenceNumber even when all events have been deleted. Sequence numbers should not be reused.

Reply:
Thanks for the hint, I will retest it with that.

> The purpose of keeping the last row is to know the latest sequenceNumber even when all events have been deleted. Sequence numbers should not be reused.

The API should be consistent across different implementations (cassandra/jdbc). Since the Cassandra plugin allows deleting all events while maintaining the sequence number in a separate table, the JDBC plugin might follow the same pattern.

Reply:
Right, that seems to get the correct behavior